### PR TITLE
Unify websockets tests for Starlite

### PR DESCRIPTION
**Pull request by kristjanvalur**
Move the tests for the websockets integration for Starlite to the new `tests/websocket` location.

## Description

Websockets tests for all integrations are unified under /tests/websockets and parametrized
so that the tests can be written once.  Starlite tests were still in their old location.

In addition, fixed some cases in the integration where "close" messages didn't contain a "reason", and
some other cleanup.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by kristjanvalur** [comment]
There is no RELEASE.md, since this is testsing only.

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @kristjanvalur for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: tests/http/clients/starlite.py [hunk 1]
```diff
@@ -129,3 +159,82 @@ async def post(
...
+            with self.client.websocket_connect(url, protocols) as ws:
+                yield StarliteWebSocketClient(ws)
+        except WebSocketDisconnect as error:
+            ws = StarliteWebSocketClient(None)
+            ws.handle_disconnect(error)
+            yield ws
+
+
+class StarliteWebSocketClient(WebSocketClient):
+    def __init__(self, ws: Any):
```
Comment (DoctorJohn) [inline tests/http/clients/starlite.py:180]: nit: AFAIK `ws` would be a `WebSocketTestSession` imported via `from startlite.testing import WebSocketTestSession`

**Review by DoctorJohn — APPROVED** [review]
LGTM couldn't spot any issue comparing it to the other test clients :+1:

**Comment by kristjanvalur** [comment]
> Also just learned that Starlite is now called Litestar lol

Yes, it just happened.  Someone must have complained about naming confusion.

**Comment by kristjanvalur** [comment]
FYI, I just submitted a PR to the django/channels to update the websockets support to ASGI spec 2.3:  https://github.com/django/channels/pull/2002

**Pull request merged**